Compiler/General.py

```diff
@@ -363,6 +363,95 @@ def get_if_equal_to_0_code(inside_if_code, offset_to_temp_cell):
     return code
 
 
+def get_bitwise_code(code_logic):
+    # a, b, c, w, x, y, z, bit1, bitcounter, res
+
+    code = ">" * 7  # point to bit1
+    code += "[-]"  # zero bit1
+    code += ">"  # point to bitcounter
+    code += ">[-]<"  # zero res
+
+    code += "[-]--------[++++++++"  # while bitcounter != 8:
```

Comment:
why add 8 and sub 7 at lin 444? can simply add 1 here :)

Reply:
It's a while loop that checks if `bit_counter != 8:`

The order is important because it takes the lowest bit from a and b, and then divides by 2 for the next iteration.

And then it uses bit_counter to `res_bit << bit_counter`

If the bit_counter is subtracted by 7 in the end and not added again in the beginning it will forget the value it was.